src/assets/TokenDetectionController.test.ts

```diff
@@ -110,29 +107,30 @@ describe('TokenDetectionController', () => {
   let network: NetworkController;
   let tokensController: TokensController;
   let tokenList: TokenListController;
-  let assetsContract: AssetsContractController;
   let getBalancesInSingleCall: sinon.SinonStub<
     Parameters<AssetsContractController['getBalancesInSingleCall']>,
     ReturnType<AssetsContractController['getBalancesInSingleCall']>
   >;
 
   beforeEach(async () => {
-    preferences = new PreferencesController();
+    nock(TOKEN_END_POINT_API)
+      .get(`/tokens/${NetworksChainId.mainnet}`)
+      .reply(200, sampleTokenList)
+      .get(
```

Comment:
Nit: Which tests were these two mock responses setup for? It would be better to co-locate these with the tests. These `beforeEach` mocks tend to obscure which tests rely upon which mocks

Reply:
I agree, I'm going to track this in another ticket though since moving this will also require some reshuffling of how the `beforeEach` code is initialized - Not a small refactor ATM.